api/types/src/convert.rs

```diff
@@ -754,7 +759,22 @@ impl<'a, R: MoveResolverExt + ?Sized> MoveConverter<'a, R> {
         type_tag: &TypeTag,
         val: Value,
     ) -> Result<move_core_types::value::MoveValue> {
-        let layout = self.inner.get_type_layout_with_types(type_tag)?;
+        let layout = match type_tag {
+            TypeTag::Struct(boxed_struct) => {
+                // This has to be checked at this level, otherwise it goes into the Move language
```

Comment:
might rephrase that the current framework cannot parse generic layouts so we perform this here